src/ORM/DB.php

```diff
@@ -109,14 +162,51 @@ public static function get_conn($name = 'default')
         return null;
     }
 
+    /**
+     * Get whether the primary database connection must be used for the rest of the current request
+     * and not a replica connection
+     */
+    public static function getMustUsePrimary(): bool
+    {
+        return DB::$mustUsePrimary;
```

Comment:
```suggestion
        return DB::$mustUsePrimary || DB::$withPrimaryCount > 0;
```
In both of these scenarios, the primary must be used. It doesn't matter _why_ the primary must be used, just that it must.

Reply:
I don't want to add this logic to the existing getter as as this this confusing as it's supposed to be getter for the property.

However since the setter kind of isn't a real setting, I'll rename this method to 'isUsingPrimary()' and add in the check for replicate config, i.e. I'll extract the logic from the top of DB::getDefaultConnectionName()

Reply:
Ahh okay. My intention was always for this to be the "tell me if I am using the primary" method, not the "Get that property's value" method.

Probably not worth having a getter just for this property on its own, that'll just be confusing I think especially when there's also a `willUsePrimary`.
IMO remove this getter

Reply:
Removed

Reply:
It's still here?

Reply:
It was removed